sdk/export/exporter.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package export
+
+import (
+	"context"
+)
+
+// SpanSyncer is a type for functions that receive a single sampled trace span.
+//
+// The ExportSpan method is called synchronously. Therefore, it should not take
+// forever to process the span.
+//
+// The SpanData should not be modified.
+type SpanSyncer interface {
+	ExportSpan(context.Context, *SpanData)
```

Comment:
This is probably a bigger questions, but how do we plan on propagating errors back from these interfaces?

Reply:
An exporter getting an error shouldn't impact other exporters in the stack. So I believe they should log their errors, but there's no point in them sending it upstream.

Reply:
Exporter should allow registering optional callback function OnError()

Reply:
That looks like it's outside the scope of this PR though? I've opened #174.

Reply:
Agree, that's out of scope. Thanks for #174.